### DISCUSSION TITLE
edge3 versus edge2

### DISCUSSION THREAD

**Discussion by dealmeidavf** [idaholab/moose#17039] [Q&A General] [answered]
This input

```python
[Mesh]
  [1d]
   type = GeneratedMeshGenerator
   dim = 1
   nx = 4
   xmin = -5
   xmax = 5
   elem_type = edge3
  []
[]

[Variables]
  [u]
    order = second
    family = lagrange
  []
[]
```

will not run unless `elem_type = edge3`. The error is:

>> ERROR: Bad ElemType = EDGE2 for SECOND order approximation!

What is the difference between `edge2` and  `edge3`.
Thanks.

**Comment by GiudGiud** [comment, marked as answer]
For second order variables you need a second order mesh. 
Specify `second_order = true` in the [Mesh] block directly then you dont need to specify the elem_type for it to work I think
The difference is the number of dofs on the edge.

**Reply by dealmeidavf** [reply]
I tried but still got

> ERROR: Bad ElemType = EDGE2 for SECOND order approximation!

It is ok. I understand the EDGE3 for quadratic Lagrange polynomials in 1D. The "edge" naming did not connect to 1D at first.
Thanks.
